### ISSUE TITLE
`email.headerregistry` docs omit the required keyword `parse_tree`

### ISSUE THREAD

**Issue by perey** [python/cpython#123138]
# Documentation

The documentation for the `email.headerregistry` module say that the `parse()` method of custom header classes must ensure that `kwds` contains the keys `decoded` and `defects`:
https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Doc/library/email.headerregistry.rst?plain=1#L99

However, there is a third key required, `parse_tree`, as mentioned in the source code:
https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Lib/email/headerregistry.py#L165

The documentation should mention this required key as well. Ideally, it would also explain what its content should be.

### Linked PRs
* gh-134450

**Comment by bitdancer** [comment]
I think that at the time I wrote that doc I was thinking that you could make a policy that didn't need the parse tree, and thus that keyword could be optional.  Obviously I didn't implement it that way, and it would be a really unusual application that did something like that.  Such an application could just as well return a dummy parse tree, or its own object to use in a similar way, so keeping the keyword as required probably makes the most sense at this point.

What the parse tree is used for is refolding the header at serialization time.  What the parse tree *is* is the token list produced by _header_value_parser, which as the leading underscore indicates is a private library...and thus its contents is not documented and should not be documented at this time.

I'm not yet ready to make the header value parser's data structure non-private, since that would put backward compatibility constraints on the objects it returns in ways that would cripple the rewrite I'm currently doing ;) (I'm trying to keep the changes minimal, but not *that* minimal.)  What we could do here in the meantime is say that the returned data structure is used by the "modern policies" (as we've been calling them) for header folding, and that it is returned by the private header value parser, making the details of its contents subject to change at any time.

I do plan that the parser and its data structure will be fully documented at some point, but it may be quite a while before I'm ready for that.  We're still dealing with bugs that require tweaking the data structure details, and the API surface it provides needs some improvement.

**Comment by serhiy-storchaka** [comment]
So, is there anything else that needs to be done in this issue?

**Comment by bitdancer** [comment]
I don't think so.  I just wanted to have the answer to the question of the content on the record somewhere ;)
